lib/puppetserver/ca/generate_action.rb

```diff
@@ -220,6 +267,10 @@ def self.parser(parsed = {})
           opts.on('--config CONF', 'Path to puppet.conf') do |conf|
             parsed['config'] = conf
           end
+          opts.on('--subject-alt-names NAMES',
```

Comment:
Is there a benefit to processing this as a string instead of Array like we do for certnames?

Reply:
also, i think we need to change the error handling to catch when we don't provide any names to the flag:
```
be ./exe/puppetserver-ca generate --subject-alt-names
Error:
Unknown arguments or flags:
    --subject-alt-names
```

Reply:
which is my fault:
```
be ./exe/puppetserver-ca generate --config
Error:
Unknown arguments or flags:
    --config
    
```

Reply:
Should be easily fixable with new parsing we've been using (replacing parse and parse_inputs) with:
```
def parse(args)
        results = {}
        parser = self.class.parser(results)

        errors = Utils.parse_with_errors(parser, args)

        if check_flag_usage(results)
          errors << check_flag_usage(results)
        end

        errors_were_handled = Utils.handle_errors(@logger, errors, parser.help)

        exit_code = errors_were_handled ? 1 : nil

        return results, exit_code
      end
```

Reply:
Oh, should "NAMES" here be  comma separated like "FOO,BAR" ?

Reply:
As far as benefit of processing as a string, I only did it that way because the value coming out of Puppet's config is a string. So alternatively I could call `split` just on that one before passing it to munge, but it kind of seemed six-a-one-half-a-dozen-the-other to me. And since they ultimately need to be a single string when the get attached to the cert, I guess I was thinking that the fact they become an array in the middle (for inserting prefixes) is mostly an implementation detail.